src/inference/src/dev/core_impl.cpp

```diff
@@ -1007,6 +1008,40 @@ ov::AnyMap ov::CoreImpl::get_supported_property(const std::string& full_device_n
     return supported_config;
 }
 
+ov::AnyMap ov::CoreImpl::get_supported_property(const std::string& full_device_name,
+                                                const ov::AnyMap& user_properties,
+                                                const bool keep_core_property) const {
+    if (is_virtual_device(full_device_name)) {
+        // Considerations:
+        // 1. in case of virtual devices all the magic will happen on the level when
+        // virtual device calls ICore::get_supported_property for real HW devices
+        // so, for now we can return user properties almost as is without any
+        // filtering / flattening
+        // 2. The only exception here: while common properties like ov::num::streams or
+        // ov::hint::performance_mode are shared across all the devices, the
+        // ov::device::priority cannot be shared, because it's specific for current virtual
+        // plugin. So, we need to remove ov::device::priorities from the list, because it's
+        // supposed to be set for current virtual plugin and cannot be propagated down
+        auto return_properties = user_properties;
+        auto device_priorities_it = return_properties.find(ov::device::priorities.name());
+        if (device_priorities_it != return_properties.end()) {
+            return_properties.erase(device_priorities_it);
+        }
+        return return_properties;
+    } else {
+        const auto parsed = ov::parseDeviceNameIntoConfig(full_device_name);
+        return get_supported_property(get_plugin(parsed._deviceName), user_properties, keep_core_property, rw_only);
+    }
+}
+
+ov::AnyMap ov::CoreImpl::get_hw_plugin_properties_or_forward(const Plugin& plugin, const AnyMap& config) const {
+    constexpr auto keep_core = false;
+    const auto& device_name = plugin.get_name();
+    const auto forward_config = is_virtual_device(device_name) || is_proxy_device(device_name);
+
+    return forward_config ? config : get_supported_property(plugin, config, keep_core, !rw_only);
```

Comment:
does it mean if I pass unsupported property to `core.compile_model(CPU, some_property(XX))`, then no exception is thrown ?

Reply:
No it should throw if device not support property.

Assume usage `core.compile_model(model, "CPU", config)` and confgi has got property:

| Property | Result (this branch) | Result (master) |
| :---          |:---                    | :---                    |
| {ov::enable_mmap(false)} | no throw (core property) | throws |
| {ov::force_tbb_terminate(false)} | no throw (core property) | throws |
| {ov::intel_gpu::enable_loop_unrolling(false)} | throws not supported by CPU (GPU prop) | throws |
| {"test", "my_value"} |  throws not supported by CPU (custom user) | throws |

I think that was expected behavior in reported ticket and should be correct.

Reply:
```c++
core.compile_model(file_path, "CPU", ov::enable_mmap(false));
```
will `read_model` with `mmap=true` anyway? as `coreConfig` will not have updated information about `ov::enable_mmap(false)`

Reply:
Yes, the `coreConfig` is not changed, as it not allowed to change Core state.

There is separate ticked to allow use core properties in user config (e.g. to read model).

Reply:
> Yes, the coreConfig is not changed, as it not allowed to change Core state.

and suggestion to create local coreCOnfig which is union of global CoreConfig and locally passed properties.